### ISSUE TITLE
Multiplayer Netplay support.

### ISSUE THREAD

**Issue by Zombie-Ryushu** [deathkiller/jazz2-native#39]
### Jazz² Resurrection version

2.2.2

### Problem or limitation

Are there any plans to add netplay support to Jazz 2 Resurrection? I would like to be able to have multiple players playing Jazz and Spaz collaboratively.

### Proposed solution

Add a peer to peer network Stack using the network Stack utilized in OpenJazz.

**Comment by deathkiller** [comment]
Yes, it's in development, there are 5 commits already. The last one (created 2 days ago) is playable but not complete, so it's not enabled in public builds.

**Comment by FrostKnight** [comment]
> Yes, it's in development, there are 5 commits already. The last one (created 2 days ago) is playable but not complete, so it's not enabled in public builds.

I look forward to that being done. :)

**Comment by deathkiller** [comment]
Online multiplayer has entered the pre-release testing phase - it's available as a separate build to download and test it.
All core multiplayer features should now work. Battle, race and co-op modes are implemented (along with the elimination option and ranking). Server can be configured via a configuration file as described [here](https://deat.tk/jazz2/help/multiplayer) and [here](https://deat.tk/jazz2/docs/structJazz2_1_1Multiplayer_1_1ServerConfiguration.html). The UI is still unpolished, but that's not necessary for testing. There are still [some known issues](https://deat.tk/jazz2/help/multiplayer#known-issues), but they don't affect general usability, so they're a lower priority.

**I'm not going to implement any new features until I have more feedback from people testing it. I want to see how the testing goes first to avoid rewriting too much code. So please let me know what do you think, what works well and what works badly, so we can move forward.** If everything does well, I can eventually enable multiplayer even in the main build.

To participate in the testing, feel free to join [Discord](https://discord.gg/Y7SBvkD), assign yourself the **<span>@</span>Testers** role there and follow the instructions in ⁠#testing-feedback. If you don't have Discord, you can download the multiplayer builds directly from GitHub Actions, but I need to hear some feedback from you, so it's highly recommended to have it.

**Comment by deathkiller** [comment]
Online multiplayer has beed released in version 3.7.0.
